sql/src/main/java/io/crate/metadata/Routing.java

```diff
@@ -18,6 +20,8 @@
     private Map<String, Map<String, List<Integer>>> locations;
     private volatile int numShards = -1;
 
+    private static Random random = new Random();
```

Comment:
Maybe replace with `ThreadLocalRandom` http://stackoverflow.com/questions/363681/how-to-generate-random-integers-within-a-specific-range-in-java/363692#363692